src/main/java/com/redislabs/redisai/Model.java

```diff
@@ -0,0 +1,215 @@
+   */
+  public Model(Backend backend, Device device, String[] inputs, String[] outputs, byte[] blob) {
+    this.backend = backend;
+    this.device = device;
+    this.inputs = inputs;
+    this.outputs = outputs;
+    this.blob = blob;
+    this.tag = null;
+    this.batchSize = 0;
+    this.minBatchSize = 0;
+  }
+
+  public static Model createModelFromRespReply(List<?> reply) {
+    Model model = null;
+    Backend backend = null;
+    Device device = null;
+    String tag = null;
+    byte[] blob = null;
+    long batchsize = -1;
+    long minbatchsize = -1;
+    String[] inputs = new String[0];
+    String[] outputs = new String[0];
+    for (int i = 0; i < reply.size(); i += 2) {
+      String arrayKey = SafeEncoder.encode((byte[]) reply.get(i));
+      switch (arrayKey) {
+        case "backend":
+          String backendString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          backend = Backend.valueOf(backendString);
+          if (backend == null) {
+            throw new JRedisAIRunTimeException("Unrecognized backend: " + backendString);
+          }
+          break;
+        case "device":
+          String deviceString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          device = Device.valueOf(deviceString);
+          if (device == null) {
+            throw new JRedisAIRunTimeException("Unrecognized device: " + deviceString);
+          }
+          break;
+        case "tag":
+          tag = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          break;
+        case "blob":
+          blob = (byte[]) reply.get(i + 1);
+          break;
+        case "batchsize":
+          batchsize = (Long) reply.get(i + 1);
+          break;
+        case "minbatchsize":
+          minbatchsize = (Long) reply.get(i + 1);
+          break;
+        case "inputs":
+          List<byte[]> inputsEncoded = (List<byte[]>) reply.get(i + 1);
+          if (inputsEncoded.size() > 0) {
+            inputs = new String[inputsEncoded.size()];
+            for (int j = 0; j < inputsEncoded.size(); j++) {
+              inputs[j] = SafeEncoder.encode(inputsEncoded.get(j));
+            }
+          }
+          break;
+        case "outputs":
+          List<byte[]> outputsEncoded = (List<byte[]>) reply.get(i + 1);
+          if (outputsEncoded.size() > 0) {
+            outputs = new String[outputsEncoded.size()];
+            for (int j = 0; j < outputsEncoded.size(); j++) {
+              outputs[j] = SafeEncoder.encode(outputsEncoded.get(j));
+            }
+          }
+          break;
+        default:
+          break;
+      }
+    }
+    if (backend != null && device != null && blob != null) {
+      model = new Model(backend, device, inputs, outputs, blob);
+      if (tag != null) {
+        model.setTag(tag);
+      }
+      if (batchsize > -1) {
+        model.setBatchSize(batchsize);
+      }
+      if (minbatchsize > -1) {
+        model.setMinBatchSize(minbatchsize);
+      }
```

Comment:
why not set those in the constructor? perhaps overload constructor with those arguments

Reply:
agree. adding it

Reply:
solved in the latest commit